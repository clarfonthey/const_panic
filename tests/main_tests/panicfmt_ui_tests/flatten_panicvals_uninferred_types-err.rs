use const_panic::{FmtArg, PanicVal, flatten_panicvals};

struct NotFmt;

const _: &[PanicVal<'_>] = &flatten_panicvals!(FmtArg::DEBUG; 0);

const _: &[PanicVal<'_>] = &flatten_panicvals!(FmtArg::DEBUG, 10; 0);

fn main() {}