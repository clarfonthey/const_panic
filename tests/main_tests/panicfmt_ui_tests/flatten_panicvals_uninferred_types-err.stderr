error[E0282]: type annotations needed
 --> tests/main_tests/panicfmt_ui_tests/flatten_panicvals_uninferred_types-err.rs:5:29
  |
5 | const _: &[PanicVal<'_>] = &flatten_panicvals!(FmtArg::DEBUG; 0);
  |                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ cannot infer type
  |
  = note: this error originates in the macro `$crate::__to_pvf_kind` which comes from the expansion of the macro `flatten_panicvals` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0282]: type annotations needed
 --> tests/main_tests/panicfmt_ui_tests/flatten_panicvals_uninferred_types-err.rs:7:29
  |
7 | const _: &[PanicVal<'_>] = &flatten_panicvals!(FmtArg::DEBUG, 10; 0);
  |                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ cannot infer type
  |
  = note: this error originates in the macro `$crate::__to_pvf_kind` which comes from the expansion of the macro `flatten_panicvals` (in Nightly builds, run with -Z macro-backtrace for more info)
